### ISSUE TITLE
check circuit_indices vs length of circuits prior to running

### ISSUE THREAD

**Issue by johannesgreiner** [Qiskit/qiskit-ibm-runtime#313]
**What is the expected feature or enhancement?**

Edit: compare length of circuits and max(circuit_indices) prior to running

**Reproduce:**
```from qiskit_ibm_runtime import QiskitRuntimeService, Sampler
from qiskit import QuantumCircuit

service = QiskitRuntimeService()

bell = QuantumCircuit(2)
bell.h(0)
bell.cx(0, 1)
bell.measure_all()

# executes three Bell circuits
with Sampler(circuits=[bell]*3, service=service, 
options={"backend": "ibmq_qasm_simulator" }) as sampler:

    result = sampler(circuit_indices=[0, 1, 2, 3], shots=400)

    print(result)
```
atm runs for <30s, then aborts w/ `RuntimeJobFailureError: 'Unable to retrieve job result`

**Acceptance criteria**

refuse to run if `max(circuit_indices) >= len(circuits)`

**Comment by HuangJunye** [comment]
Actually the list of indices does not need to be smaller than the list of circuits. What we need to make sure is that the index inside the list does not reference outside of the list of the circuit. This sentence might be a bit confusing. What I meant is that the following is allowed:

```
circuit_indices = [0,0,1,1,2,2]
```
but this causes problem

```
circuit_indices = [3]
```

**Comment by HuangJunye** [comment]
But I agree with you that some checking before submitting the job would be nicer than waiting for the job to run (which may take a while as we get more users) and get an error after that.

**Comment by johannesgreiner** [comment]
thank you, yep you're right. that's why I ended up proposing this as a check: refuse to run if  
`max(circuit_indices) >= len(circuits)`

**Comment by johannesgreiner** [comment]
thank you! good to know (possibly for external users as well). Agreed! The combination of circuits and parameters (vs observables) is beneficial for the Sampler primitive though? c.f. the Sampler tutorial. Is that part of your plan already?

**Comment by johannesgreiner** [comment]
@nbronn ^ maybe you'll need to update the [youtube ](https://www.youtube.com/watch?v=b9mdMye-iVk) on indices by Q2 ;)

**Comment by rathishcholarajan** [comment]
This has been handled as part of terra PR https://github.com/Qiskit/qiskit-terra/pull/8065 and should be available in qiskit-ibm-runtime 0.5.0 scheduled for end of June.
